Mathlib/Algebra/EuclideanDomain/Field.lean

```diff
@@ -24,3 +28,14 @@ instance (priority := 100) Field.toEuclideanDomain {K : Type*} [Field K] : Eucli
       (Acc.intro _) fun _ ⟨hb, _⟩ => (Acc.intro _) fun _ ⟨_, hnb⟩ => False.elim <| hnb hb,
   remainder_lt := fun a b hnb => by simp [hnb],
   mul_left_not_lt := fun _ _ hnb ⟨hab, hna⟩ => Or.casesOn (mul_eq_zero.1 hab) hna hnb }
+
+@[simp]
+protected theorem mod_eq (a b : K) : a % b = a - a * b / b := rfl
+
+@[simp]
+protected theorem gcd_eq [DecidableEq K] (a b : K) :
+    EuclideanDomain.gcd a b = if a = 0 then b else a := by
+  unfold EuclideanDomain.gcd
+  split_ifs <;> simp [*, Field.mod_eq]
```

Comment:
Maybe you also want `gcd 0 b = b` and `gcd a b = a` if `a \neq 0` (these two does not require `DecidableEq`)

Reply:
Calling `EuclideanDomain.gcd` itself already requires `DecidableEq`, I'm afraid. (Maybe it shouldn't, but not touching that here.)

Reply:
Ahh, I'm sorry, `gcd 0 b = b` is already [EuclideanDomain.gcd_zero_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/EuclideanDomain/Defs.html#EuclideanDomain.gcd_zero_left)